pandas/_libs/writers.pyx

```diff
@@ -128,19 +128,28 @@ def max_len_string_array(pandas_string[:] arr) -> Py_ssize_t:
 
     for i in range(length):
         val = arr[i]
-        if isinstance(val, str):
-            l = PyString_GET_SIZE(val)
-        elif isinstance(val, bytes):
-            l = PyBytes_GET_SIZE(val)
-        elif isinstance(val, unicode):
-            l = PyUnicode_GET_SIZE(val)
+        l = max_len_string(val)
 
         if l > m:
             m = l
 
     return m
 
 
+def max_len_string(val):
```

Comment:
this needs to be ``cdef inline Py_ssize_t max_len_string(object val)``

Reply:
Done! Yesterday I tried

`cdef Py_ssize_t max_len_string(object val)`
`cdef Py_ssize_t max_len_string(pandas_string val)`
`cpdef max_len_string(object val) -> Py_ssize_t` and 
`cpdef max_len_string(object val) -> Py_ssize_t`

And ended up giving up and reverting to the simpler signature above. I think `pytables` tests were failing in my IDE, but I didn't try on Travis CI. Pushed this change now to see if that passes the tests.

Thanks a ton @jreback ! Still need to run the asv's you mentioned above.

Reply:
Same error I had @jreback : https://travis-ci.org/kinow/pandas/jobs/488681016

```
AttributeError: module 'pandas._libs.writers' has no attribute 'max_len_string'
```

Is it because the `cdef` function cannot be called from Python? ([this](https://notes-on-cython.readthedocs.io/en/latest/function_declarations.html) is what I used yesterday trying to understand why I had this error). Then after that I tried `cpdef` and changed where the return type was defined, but that resulted in another error..

Reply:
correct, you can mark it cpdef to get both python and cython functionaility

Reply:
You nailed it @jreback ! It worked! 👏 🎉